File: aoc2016/aoc2016-java/src/test/java/de/havox_design/aoc2016/day17/Day17Test.java
package de.havox_design.aoc2016.day17;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day17Test {

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, String expected) {
        Assertions.assertEquals(expected, TwoStepsForward.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample1.txt", "DDRRRD"),
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample2.txt", "DDUDRLRRUDRD"),
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample3.txt", "DRURDRUDDLLDLUURRDULRLDUUDDDRR")
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForPart2")
    void testPart2(String fileName, long expected) {
        Assertions.assertEquals(expected, TwoStepsForward.solvePart2(fileName));
    }

    private static Stream<Arguments> getDataForPart2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample1.txt", 370L),
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample2.txt", 492L),
                Arguments.of("de/havox_design/aoc2016/day17/day17Sample3.txt", 830L)
        );
    }
}

File: aoc2023/aoc2023-kotlin/src/test/kotlin/de/havox_design/aoc2023/day17/Day17Test.kt
package de.havox_design.aoc2023.day17

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day17Test {
    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart1")
    fun testSolvePart1(filename: String, expectedResult: Long) =
        ClumsyCrucible(filename).solvePart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart2")
    fun testSolvePart2(filename: String, expectedResult: Long) =
        ClumsyCrucible(filename).solvePart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestSolvePart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2023/day17/day17Sample1.txt", 102L),
                Arguments.of("de/havox_design/aoc2023/day17/day17Sample2.txt", 59L)
            )

        @JvmStatic
        private fun getDataForTestSolvePart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2023/day17/day17Sample1.txt", 94L),
                Arguments.of("de/havox_design/aoc2023/day17/day17Sample2.txt", 71L)
            )
    }
}
